contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,559 @@
+# Local Raw Block Consumption in Kubernetes
+
+Authors: erinboyd@, screeley44@, mtanino@
+
+This document presents a proposal for managing raw block storage in Kubernetes using the persistent volume source API as a consistent model of consumption.
+
+# Terminology
+* Raw Block Device - a physically attached device devoid of a filesystem
+* Raw Block Volume - a logical abstraction of the raw block device as defined by a path
+* File on Block - a formatted (ie xfs) filesystem on top of a raw block device
+
+# Goals
+* Enable durable access to block storage
+* Support storage requirements for all workloads supported by Kubernetes
+* Provide flexibility for users/vendors to utilize various types of storage devices
+* Agree on API changes for block
+* Provide a consistent security model for block devices 
+* Provide a means for running containerized block storage offerings as non-privileged container
+
+# Non Goals
+* Support all storage devices natively in upstream Kubernetes. Non-standard storage devices are expected to be managed using extension
+  mechanisms.
+   
+# Value add to Kubernetes
+
+  Before the advent of storage plugins, emptyDir and hostPath were widely used to quickly prototype stateless applications in Kubernetes. 
+  Both have limitations for their use in application that need to store persistent data or state. 
+  EmptyDir, though quick and easy to use, provides no real guarantee of persistence for suitable amount of time. 
+  Appropriately used as scratch space, it does not have the HostPath, became an initial offering for local storage, but had many
+  drawbacks. Without having the ability to guarantee space & ensure ownership, one would lose data once a node was rescheduled. 
+  Therefore, the risk outweighed the reward when trying to  leverage the power of local storage needed for stateful applications like 
+  databases.
+  
+  By extending the API for volumes to specifically request a raw block device, we provide a consist method for accessing volumes. In
```

Comment:
By having the raw block devices created as PVs they are more consumable rather than in terms of how file on block is handled today, the raw devices are created outside of kube and kube has no knowledge or reference to what is available.